modules/camera/camera_android.cpp

```diff
@@ -291,74 +291,170 @@ void CameraFeedAndroid::onImage(void *context, AImageReader *p_reader) {
 	// Get image
 	AImage *image = nullptr;
 	media_status_t status = AImageReader_acquireNextImage(p_reader, &image);
-	ERR_FAIL_COND(status != AMEDIA_OK);
+	if (status != AMEDIA_OK) {
+		if (status == AMEDIA_IMGREADER_NO_BUFFER_AVAILABLE) {
+			// No buffer available - normal condition under high load, skip silently
+			return;
+		} else {
+			print_error(vformat("AImageReader_acquireNextImage failed with status: %d", status));
+			return;
+		}
+	}
 
 	// Get image data
 	uint8_t *data = nullptr;
 	int len = 0;
-	int32_t pixel_stride, row_stride;
+	int32_t pixel_stride;
 	FeedFormat format = feed->get_format();
 	int width = format.width;
 	int height = format.height;
 	switch (format.pixel_format) {
-		case AIMAGE_FORMAT_YUV_420_888:
+		case AIMAGE_FORMAT_YUV_420_888: {
+			// Get Y plane data and stride
+			int32_t y_row_stride;
+			AImage_getPlaneRowStride(image, 0, &y_row_stride);
 			AImage_getPlaneData(image, 0, &data, &len);
 			if (len <= 0) {
 				return;
 			}
 			if (len != data_y.size()) {
-				int64_t size = Image::get_image_data_size(width, height, Image::FORMAT_R8, false);
-				data_y.resize(len > size ? len : size);
+				data_y.resize(len);
 			}
 			memcpy(data_y.ptrw(), data, len);
 
+			// Get UV plane data and stride
+			int32_t uv_row_stride;
 			AImage_getPlanePixelStride(image, 1, &pixel_stride);
-			AImage_getPlaneRowStride(image, 1, &row_stride);
+			AImage_getPlaneRowStride(image, 1, &uv_row_stride);
 			AImage_getPlaneData(image, 1, &data, &len);
 			if (len <= 0) {
 				return;
 			}
 			if (len != data_uv.size()) {
-				int64_t size = Image::get_image_data_size(width / 2, height / 2, Image::FORMAT_RG8, false);
-				data_uv.resize(len > size ? len : size);
+				data_uv.resize(len);
 			}
 			memcpy(data_uv.ptrw(), data, len);
 
-			image_y->initialize_data(width, height, false, Image::FORMAT_R8, data_y);
-			image_uv->initialize_data(width / 2, height / 2, false, Image::FORMAT_RG8, data_uv);
+			// Process Y plane - handle stride by creating a packed buffer
+			Vector<uint8_t> packed_y;
+			packed_y.resize(width * height);
+			if (y_row_stride == width) {
+				// No padding, direct copy
+				memcpy(packed_y.ptrw(), data_y.ptr(), width * height);
```

Comment:
We are copying the same data twice here, is there any chances that this can be improved?